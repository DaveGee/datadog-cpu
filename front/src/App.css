.App {
  margin: 20px;
  font-variant: small-caps;
  font-family: 'Roboto', sans-serif;
  min-width: 1000px;
}

.App h1 {
  font-weight: normal;
  font-size: 3em;
  margin: 0;
}