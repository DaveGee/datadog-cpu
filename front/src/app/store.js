import { configureStore } from '@reduxjs/toolkit'
import loadReducer from '../features/cpuLoad/loadSlice'

const reducer = {
  cpuLoad: loadReducer
}

const buildStore = preloadedState => configureStore({
  reducer,
  devTools: process.env.NODE_ENV !== 'production',
  preloadedState,
})

export const store = buildStore()
export const getStore = preloadedState => buildStore(preloadedState)