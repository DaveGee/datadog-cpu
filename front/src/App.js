import React from 'react'
import LoadMetric from './features/loadMetric/LoadMetric'
import './App.css'

function App() {
  return (
    <div className="App">
      <LoadMetric currentLoad={0.5543} />
    </div>
  )
}

export default App
