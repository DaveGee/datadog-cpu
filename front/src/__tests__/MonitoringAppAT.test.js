import React from 'react'
import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import { store } from '../app/store'
import App from '../App'

describe('Monitoring app should', () => {
  test('display a cpu load chart', () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>
    )

    expect(screen.getByText('CPU Load: 10 minute window')).toBeInTheDocument()
  })
})

