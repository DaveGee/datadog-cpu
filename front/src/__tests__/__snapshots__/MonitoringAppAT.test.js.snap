// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monitoring app should render consistently 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Datadog CPU load monitoring
    </h1>
    <div
      class="loadChartContainer"
    >
      <div
        class="metrics"
      >
        <dl
          class="events"
        >
          <dt>
            Alerts
          </dt>
          <dd>
            0
          </dd>
          <dt>
            Recoveries
          </dt>
          <dd>
            0
          </dd>
        </dl>
        <dl
          class="loadMetric"
        >
          <dt>
            Current load
          </dt>
          <dd>
            0
          </dd>
        </dl>
      </div>
      <div
        class="loadChart"
      >
        <div
          height="100%"
          width="100%"
        />
      </div>
      <div>
        <button>
          Simulate
        </button>
      </div>
    </div>
  </div>
</div>
`;
